docs/content/en/blog/news/primary-cache-for-next-recon.md

```diff
@@ -0,0 +1,83 @@
+---
+
+We recently released v5.1 of Java Operator SDK. One of the highlights of this release is related to a topic of so-called
+[allocated values](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#representing-allocated-values
+).
+
+To describe the problem, let's say if we create a resource from our controller that has a generated ID 
+we have to store this ID, usually in the `.status` of the custom resource. 
+(In other words, we cannot address the resource only by using the values from the `.spec`.) 
+However, operator frameworks cache resources
+using informers, so the update that you made to the status of the custom resource will just eventually get into 
+the cache of the informer. If meanwhile some other event triggers the reconciliation, it can happen that we will 
+see the stale custom resource in the cache (in another word, the cache is eventually consistent) without the generated ID in the status. 
+This is a problem since we might not know at that point that the desired resources were already created, so it might happen that you try to 
+create them again. 
+
+Java Operator SDK now out of the box provides a utility class [`PrimaryUpdateAndCacheUtils`](https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/PrimaryUpdateAndCacheUtils.java)
+if you use it, the framework guarantees that the next reconciliation will always receive the updated resource:
+
+```java
+ @Override
+  public UpdateControl<StatusPatchCacheCustomResource> reconcile(
+      StatusPatchCacheCustomResource resource,
+      Context<StatusPatchCacheCustomResource> context) {
+    
+    // omitted code
+    
+    var freshCopy = createFreshCopy(resource); // need fresh copy just because we use the SSA version of update
+    freshCopy
+        .getStatus()
+        .setValue(statusWithAllocatedValue());
+
+    // using the utility instead of update control
+    var updated =
+        PrimaryUpdateAndCacheUtils.ssaPatchStatusAndCacheResource(resource, freshCopy, context);
+    return UpdateControl.noUpdate();
+  }
+```
+
+This utility class will do the magic for you. But how does it work? 
+There are multiple ways to solve this problem, 
+but ultimately, we only provided the solution below. (If you want to dig deep in alternatives, see this [PR](https://github.com/operator-framework/java-operator-sdk/pull/2800/files)).
+
+The trick is to cache the resource from the response of our update in an additional cache on top of the informer's cache.
+If we read the resource, we first check if it is in the overlay cache and read it from there if present, otherwise read it from the cache of the informer. 
+If the informer receives an event with a fresh resource, we always remove the resource from the overlay 
+cache, since that is a more recent resource. But this **works only** if the update is done **with optimistic locking**.
```

Comment:
What does "update" refer to here? This isn't clear.

Reply:
improved wording ot it